File: 205.同构字符串.cpp
class Solution {
public:
    bool isIsomorphic(string s, string t) {
        // map<char, int> m1;
        // map<char, int> m2;
        // for(int i = 0; i < s.length(); ++i){
        //     m1[s[i]] = INT_MIN;
        //     m2[t[i]] = INT_MIN;
        // }
        // int flag = 1;
        // for(int i = 0; i < s.length(); ++i){
        //     int temp = s[i] - t[i];
        //     if(m1[s[i]] == INT_MIN && m2[t[i]] == INT_MIN){
        //         m1[s[i]] = temp;
        //         m2[t[i]] = temp;
        //     }
        //     else if(m1[s[i]] != temp || m2[t[i]] != temp){
        //         flag = 0;
        //         break;
        //     }
        // }
        // return flag;
        int i = 0;
        while(i++ < s.length()){
            if(s.find(s[i]) != t.find(t[i]))
                return false;
        }
        return true;
    }
};

File: 205.同构字符串.java
class Solution {
    public boolean isIsomorphic(String s, String t) {
        int n = s.length();
        if(n == 0)
            return true;
        Map<Character, Character> map = new HashMap<Character, Character>();
        Map<Character, Integer> map2 = new HashMap<Character, Integer>();
        for(int i = 0; i < n; ++i){
            char c1 = s.charAt(i);
            char c2 = t.charAt(i);
            if(map.get(c1) == null){
                if(map2.get(c2) != null)
                    return false;
                map.put(c1, c2);
                map2.put(c2, 1);
            }
            else {
                char temp = map.get(c1);
                if(temp != c2)
                    return false;
            }
        }
        return true;
    }
}